Tests/Docker/Runner/DataLoader/StagingWorkspaceFactoryTest.php

```diff
@@ -0,0 +1,314 @@
+    {
+        yield 'snowflake-bigquery' => [
+            'inputStaging' => 'workspace-snowflake',
+            'outputStaging' => 'workspace-bigquery',
+            'expectedError' =>
+                'Component staging setting mismatch - input: "workspace-snowflake", output: "workspace-bigquery".',
+        ];
+
+        yield 'bigquery-snowflake' => [
+            'inputStaging' => 'workspace-bigquery',
+            'outputStaging' => 'workspace-snowflake',
+            'expectedError' =>
+                'Component staging setting mismatch - input: "workspace-bigquery", output: "workspace-snowflake".',
+        ];
+    }
+
+    /** @dataProvider provideNewWorkspaceTestData */
+    public function testCreateNewWorkspace(
+        array $tokenFlags,
+        StagingType $componentStaging,
+        array $configData,
+        NewWorkspaceConfig $expectedWorkspaceConfig,
+    ): void {
+        $storageApiToken = new StorageApiToken([
+            'owner' => $tokenFlags,
+        ], 'token-value');
+
+        $component = new ComponentSpecification([
+            'id' => 'test-component',
+            'data' => [
+                'definition' => [
+                    'type' => 'dockerhub',
+                    'uri' => 'test/test',
+                ],
+                'staging-storage' => [
+                    'input' => $componentStaging->value,
+                    'output' => $componentStaging->value,
+                ],
+            ],
+        ]);
+
+        // this is just simplification for the test, in production the backendType is produced by Connection
+        // Connection backend has the same name as our StagingType, just without the "workspace-" prefix
+        $backendType = preg_replace('/^workspace-/', '', $componentStaging->value);
+
+        $createdWorkspace = new WorkspaceWithCredentials(
+            new Workspace(
+                id: 'workspace-id',
+                backendType: $backendType,
+                backendSize: 'small',
+                loginType: WorkspaceLoginType::DEFAULT,
+            ),
+            [
+                'password' => '<PASSWORD>',
+            ],
+        );
+
+        $workspaceProvider = $this->createMock(WorkspaceProvider::class);
+        $workspaceProvider->expects(self::once())
+            ->method('createNewWorkspace')
+            ->with(
+                $storageApiToken,
+                $expectedWorkspaceConfig,
+            )
+            ->willReturn($createdWorkspace)
+        ;
+
+        $factory = new StagingWorkspaceFactory(
+            $workspaceProvider,
+            $this->logger,
+        );
+
+        $stagingWorkspaceFacade = $factory->createStagingWorkspaceFacade(
+            $storageApiToken,
+            $component,
+            Configuration::fromArray($configData),
+            null,
+        );
+
+//        self::assertTrue($this->logsHandler->hasNotice('Creating a new ephemeral workspace.'));
```

Comment:
proč je to zakomentovaný?

Reply:
Protoze v jednom pripade je ta message trochu jina a zapomnel jsme to doplnit :+1: